src/lighteval/models/vllm/vllm_model.py

```diff
@@ -411,3 +420,195 @@ def loglikelihood_rolling():
+
+        if isinstance(request, LoglikelihoodRequest):
+            sampling_params.temperature = 0
+            sampling_params.prompt_logprobs = 1
+            sampling_params.max_tokens = 1
+            sampling_params.detokenize = False
+            prompt = request.context + request.choice
+            index = f"logprob_{index}"
+        elif isinstance(request, GreedyUntilRequest):
+            sampling_params.n = request.num_samples
+            sampling_params.max_tokens = self._config.generation_parameters.max_new_tokens or request.generation_size
+            sampling_params.stop = [] if self.use_chat_template else request.stop_sequence
+            sampling_params.logprobs = int(request.use_logits)
+            prompt = request.context
+            index = f"generative_{index}"
+
+        output = await anext(
+            self.model.generate(request_id=str(index), prompt=prompt, sampling_params=sampling_params)
+        )
+        return output
+
+    async def _async_batch(self, requests: list[GreedyUntilRequest | LoglikelihoodRequest]) -> list:
+        # if self.model.engine_core.is_sleeping_async():
+        #    await self.model.engine_core.wake_up_async()
+        processed_requests = [
+            self._async_one_item(index=index, request=request) for index, request in enumerate(requests)
+        ]
+        results = await asyncio.gather(*processed_requests)
+        # await self.model.engine_core.sleep_async()
+        return results
+
+    async def greedy_until(
+        self,
+        requests: list[GreedyUntilRequest],
+        override_bs: Optional[int] = None,
+    ) -> list[GenerativeResponse]:
+        """
+        Generates responses using a greedy decoding strategy until certain ending conditions are met.
+
+        Args:
+            requests (list[Request]): list of requests containing the context and ending conditions.
+            override_bs (int, optional): Override the batch size for generation. Defaults to None.
+
+        Returns:
+            list[GenerateReturn]: list of generated responses.
+        """
+        for request in requests:
+            request.stop_sequence = as_list(request.stop_sequence) + [self.tokenizer.eos_token]
+            request.tokenized_context = self.tok_encode(request.context)
+
+        results = []
+
+        responses: list[RequestOutput] = await self._async_batch(requests=requests)
+
+        for response in responses:
+            output_token_ids = [outputs.token_ids for outputs in response.outputs]
+            full_logprobs = [output.logprobs for output in response.outputs] or []
+            logprobs = [logprob[token_id].logprob for token_id, logprob in zip(output_token_ids[0], full_logprobs[0])]
+            result = [
+                output.text for output in response.outputs
+            ]  # [logprob[token_id].decoded_token for token_id, logprob in zip(output_token_ids[0], full_logprobs[0])]
```

Comment:
Dead code?

Reply:
Yep I've been playing around storing the tokens directly vs the text - I'm still not 100% sure which way is best